packages/components/src/List/ListComposition/SortBy/SortBy.component.test.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { mount } from 'enzyme';
+import toJson from 'enzyme-to-json';
+
+import SortBy from './SortBy.component';
+import { ListContext } from '../context';
+import getDefaultT from '../../../translate';
+
+describe('SortBy', () => {
+	const defaultProps = {
+		options: [{ key: 'firstName', label: 'First Name' }, { key: 'lastName', label: 'Last Name' }],
+	};
+
+	const defaultContext = {
+		sortParams: {},
+		setSortParams: jest.fn(),
+		t: getDefaultT(),
+	};
+
+	it('should render sort by component', () => {
+		// when
+		const wrapper = mount(
+			<ListContext.Provider value={defaultContext}>
+				<SortBy id="mySortBy" {...defaultProps} />
+			</ListContext.Provider>,
+		);
+
+		// then
+		expect(toJson(wrapper)).toMatchSnapshot();
+	});
+
+	it('should render sort by component with sorting parameter from context', () => {
+		// given
+		const contextValue = { ...defaultContext, sortParams: { sortBy: 'firstName' } };
+
+		// when
+		const wrapper = mount(
+			<ListContext.Provider value={contextValue}>
+				<SortBy id="mySortBy" {...defaultProps} />
+			</ListContext.Provider>,
+		);
+
+		// then
+		expect(toJson(wrapper)).toMatchSnapshot();
```

Comment:
Same here, can you avoid snapshots except for the one testing rendering please ?

Here you need to check initial value, interact, then check resulting value. What I call value is the text in the dropdown trigger and the text in the sort order toggle.

And for controlled test, check that rendered value reflects the props, interact, and check that the callbacks are called